testing/knot-resolver4/stats-module-memset-upstreams.patch

```diff
@@ -0,0 +1,10 @@
+--- a/modules/stats/stats.c
++++ b/modules/stats/stats.c
+@@ -498,6 +498,7 @@
+ 	}
+ 	for (size_t i = 0; i < UPSTREAMS_COUNT; ++i) {
+ 		struct sockaddr *sa = (struct sockaddr *)&data->upstreams.q.at[i];
++		memset(sa, 0, sizeof(struct sockaddr));
```

Comment:
I'm by no means an expert, but why would it be necessary / make sense to zero-ize `sa` after assigning it a value? Could you not just ignore the assignment and immediately set sa to 0?

Reply:
Assigning the sa pointer to poiont at the item in the `q` array can not be skipped. We want to zero-ize the `sockaddr` struct in the array before we attempt to set a field in that struct.

Reply:
Ok, thanks for the explanation.

Reply:
You're welcome. As to why this seems to be needed for us and not others, I am not clear on that.